vortex-array/src/array/constant/compute/mod.rs

```diff
@@ -0,0 +1,146 @@
+        lhs: &ArrayData,
+        rhs: &ArrayData,
+    ) -> Option<&dyn BinaryBooleanFn<ArrayData>> {
+        // We only need to deal with this if both sides are constant, otherwise other arrays
+        // will have handled the RHS being constant.
+        (lhs.is_constant() && rhs.is_constant()).then_some(self)
+    }
+
+    fn filter_fn(&self) -> Option<&dyn FilterFn<ArrayData>> {
+        Some(self)
+    }
+
+    fn scalar_at_fn(&self) -> Option<&dyn ScalarAtFn<ArrayData>> {
+        Some(self)
+    }
+
+    fn slice_fn(&self) -> Option<&dyn SliceFn<ArrayData>> {
+        Some(self)
+    }
+
+    fn take_fn(&self) -> Option<&dyn TakeFn<ArrayData>> {
+        Some(self)
+    }
+}
+
+impl ScalarAtFn<ConstantArray> for ConstantEncoding {
+    fn scalar_at(&self, array: &ConstantArray, _index: usize) -> VortexResult<Scalar> {
+        Ok(array.owned_scalar())
+    }
+}
+
+impl TakeFn<ConstantArray> for ConstantEncoding {
+    fn take(
+        &self,
+        array: &ConstantArray,
+        indices: &ArrayData,
+        _options: TakeOptions,
+    ) -> VortexResult<ArrayData> {
+        Ok(ConstantArray::new(array.owned_scalar(), indices.len()).into_array())
+    }
+}
+
+impl SliceFn<ConstantArray> for ConstantEncoding {
+    fn slice(&self, array: &ConstantArray, start: usize, stop: usize) -> VortexResult<ArrayData> {
+        Ok(ConstantArray::new(array.owned_scalar(), stop - start).into_array())
+    }
+}
+
+impl FilterFn<ConstantArray> for ConstantEncoding {
+    fn filter(&self, array: &ConstantArray, mask: FilterMask) -> VortexResult<ArrayData> {
+        Ok(ConstantArray::new(array.owned_scalar(), mask.true_count()).into_array())
+    }
+}
+
+impl SearchSortedFn for ConstantArray {
+    fn search_sorted(&self, value: &Scalar, side: SearchSortedSide) -> VortexResult<SearchResult> {
+        match self
+            .scalar_value()
+            .partial_cmp(value.value())
+            .unwrap_or(Ordering::Less)
+        {
+            Ordering::Greater => Ok(SearchResult::NotFound(0)),
+            Ordering::Less => Ok(SearchResult::NotFound(self.len())),
+            Ordering::Equal => match side {
+                SearchSortedSide::Left => Ok(SearchResult::Found(0)),
+                SearchSortedSide::Right => Ok(SearchResult::Found(self.len())),
+            },
+        }
+    }
+}
+
+impl MaybeCompareFn for ConstantArray {
+    fn maybe_compare(
+        &self,
+        other: &ArrayData,
+        operator: Operator,
+    ) -> Option<VortexResult<ArrayData>> {
+        other.as_constant().map(|const_scalar| {
+            let lhs = self.owned_scalar();
+            let scalar = scalar_cmp(&lhs, &const_scalar, operator);
+            Ok(ConstantArray::new(scalar, self.len()).into_array())
+        })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::array::constant::ConstantArray;
+    use crate::compute::{search_sorted, SearchResult, SearchSortedSide};
+    use crate::IntoArrayData;
+
+    #[test]
+    pub fn search() {
```

Comment:
IMO, search with a null constant array is a valuable test.

Reply:
https://github.com/spiraldb/vortex/issues/1412